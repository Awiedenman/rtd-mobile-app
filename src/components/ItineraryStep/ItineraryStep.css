.transit-leg-container {
  background: white;
  display: grid;
  grid-template-columns: 40% 1% 60%;
  height: 90px;
}

.step-time-data {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.depart-time {
  font-size: 90%;
  font-weight: 600;
  margin: 0;
}

.arrive-time {
  font-size: 90%;
  font-weight: 300;
  margin: 0;
}

.step-locations-container {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: start;
  padding-left: 5px;
}

.line-info-container {
  display: flex;
  align-items: center;
}

.instructions-container {
  text-align: start;
  background: rgb(165, 163, 163);
  border: 1px black;
  padding-left: 5px;
  box-shadow: inset 0 0 0.5px 0.5px rgba(0, 0, 0, 0.5);
}

.depart-stop {
  font-size: 100%;
  font-weight: 600;
  margin: 0;
}

.transit-icon{
  height: 20px;
  width: auto;
}

.arrive-stop {
  font-size: 100%;
  font-weight: 300;
  margin: 0;
}

.instructions-container {
  display: flex;
  align-items: center;
}

.walk-icon {
  height: 14px;
  width: auto;
  margin: 5px;
}

.instructions {
  margin: 0;
  font-size: 80%;
  font-weight: 300;
}