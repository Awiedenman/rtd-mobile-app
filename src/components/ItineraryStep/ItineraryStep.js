import React from 'react';
import PropTypes from 'prop-types';

import './ItineraryStep.css';

export const ItineraryStep = ({handleClick, data}) => {

  const {
    arrival_stop,
    arrival_time,
    color,
    departure_stop,
    departure_time,
    headsign,
    imgUrl,
    instructions,
    line
  } = data;

  const styles = {
    background: color
  };

  if (headsign) {
    return (
      <div 
        className='transit-leg-container'
        onClick={handleClick}
      >
        <div className='step-time-data'>
          <p 
            className='depart-time'
          >
            {departure_time}
          </p>
          <p 
            className='arrive-time'
          >
            {arrival_time}
          </p>
        </div>
        <hr />
        <div className='step-locations-container'>
          <h3 
            className='depart-stop'
          >
            {departure_stop}
          </h3>
          <div className='line-info-container'>
            <img 
              alt='transit icon'
              className='transit-icon'
              src={`${imgUrl}`} 
            />
            <div 
              className='line-id-container'
              style={styles}
            >
              <p className='line-id'>
                {line}
              </p>
            </div>
          </div>
          <h3 
            className='arrive-stop'
          >
            {arrival_stop}
          </h3>
        </div>
      </div>
    );
  } else {
    return (
      <div className='walk-leg-container'>
        <div className='instructions-container'>
          <img
            alt='walk icon'
            className='walk-icon'
            src={`${imgUrl}`} 
          />
          <p className='instructions'>
            {instructions}
          </p>
        </div>
      </div>
    );
  }
};

export default ItineraryStep;

ItineraryStep.propTypes = {
  arrival_stop: PropTypes.string,
  arrival_time: PropTypes.string,
  color: PropTypes.string,
  departure_stop: PropTypes.string,
  departure_time: PropTypes.string,
  headsign: PropTypes.string,
  imgUrl: PropTypes.string,
  instructions: PropTypes.string,
  line: PropTypes.string,
  handleClick: PropTypes.func
}