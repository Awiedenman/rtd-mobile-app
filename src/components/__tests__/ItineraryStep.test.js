import React from 'react';
import { shallow } from 'enzyme';
import { ItineraryStep } from '../ItineraryStep/ItineraryStep';

describe('Itinerary', () => {
  let wrapper;
  let mockProps;
  beforeEach(() => {
  mockProps = {
    key: 1,
    itinerary: {
      arrival_time: "6:33am",
      departure_time: "5:51am",
      distance: "8.1 mi",
      duration: "42 mins",
      end_address: "Columbine St, Denver, CO, USA",
      favorite: false,
      itinerary_id: 59,
      start_address: "S Monaco Pkwy, Colorado, USA",
      steps: [{
        arrival_stop: "Monaco Pkwy & 29th Ave",
        arrival_time: "6:07am",
        color: "#0277bd",
        credit_name: "Regional Transportation District",
        credit_url: "http://rtd-denver.com/",
        departure_stop: "S Monaco Pkwy & Evans Ave",
        departure_time: "5:51am",
        distance: "5.5 mi",
        duration: "16 mins",
        headsign: "Central Park 65 via Monaco",
        id: 656,
        instructions: "Bus towards Central Park 65 via Monaco",
        name: "Monaco Parkway",
        num_stops: 24,
        possible_route_id: 169,
        short_name: "65",
        travel_mode: "TRANSIT",
        vehicle_type: "Bus",
      }],
      mockHandleClick: jest.fn()
    }
  }
    wrapper = shallow(
    <ItineraryStep 
      data={mockProps.itinerary.steps[0]} 
      handleClick={mockProps.mockHandleClick} 
    />)
  })

  describe('ItineraryStep', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })  
  })
  
})
