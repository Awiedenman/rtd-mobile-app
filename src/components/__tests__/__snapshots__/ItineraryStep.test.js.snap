// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary ItineraryStep should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    data={
        Object {
            "arrival_stop": "Monaco Pkwy & 29th Ave",
            "arrival_time": "6:07am",
            "color": "#0277bd",
            "credit_name": "Regional Transportation District",
            "credit_url": "http://rtd-denver.com/",
            "departure_stop": "S Monaco Pkwy & Evans Ave",
            "departure_time": "5:51am",
            "distance": "5.5 mi",
            "duration": "16 mins",
            "headsign": "Central Park 65 via Monaco",
            "id": 656,
            "instructions": "Bus towards Central Park 65 via Monaco",
            "name": "Monaco Parkway",
            "num_stops": 24,
            "possible_route_id": 169,
            "short_name": "65",
            "travel_mode": "TRANSIT",
            "vehicle_type": "Bus",
          }
    }
    handleClick={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="step-time-data"
>
          <p
                    className="depart-time"
          >
                    5:51am
          </p>
          <p
                    className="arrive-time"
          >
                    6:07am
          </p>
</div>,
        <hr />,
        <div
          className="step-locations-container"
>
          <h3
                    className="depart-stop"
          >
                    S Monaco Pkwy & Evans Ave
          </h3>
          <div
                    className="line-info-container"
          >
                    <img
                              alt="transit icon"
                              className="transit-icon"
                              src="undefined"
                    />
                    <div
                              className="line-id-container"
                              style={
                                        Object {
                                                  "background": "#0277bd",
                                                }
                              }
                    >
                              <p
                                        className="line-id"
                              />
                    </div>
          </div>
          <h3
                    className="arrive-stop"
          >
                    Monaco Pkwy & 29th Ave
          </h3>
</div>,
      ],
      "className": "transit-leg-container",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="depart-time"
>
              5:51am
</p>,
            <p
              className="arrive-time"
>
              6:07am
</p>,
          ],
          "className": "step-time-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5:51am",
              "className": "depart-time",
            },
            "ref": null,
            "rendered": "5:51am",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "6:07am",
              "className": "arrive-time",
            },
            "ref": null,
            "rendered": "6:07am",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="depart-stop"
>
              S Monaco Pkwy & Evans Ave
</h3>,
            <div
              className="line-info-container"
>
              <img
                            alt="transit icon"
                            className="transit-icon"
                            src="undefined"
              />
              <div
                            className="line-id-container"
                            style={
                                          Object {
                                                        "background": "#0277bd",
                                                      }
                            }
              >
                            <p
                                          className="line-id"
                            />
              </div>
</div>,
            <h3
              className="arrive-stop"
>
              Monaco Pkwy & 29th Ave
</h3>,
          ],
          "className": "step-locations-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "S Monaco Pkwy & Evans Ave",
              "className": "depart-stop",
            },
            "ref": null,
            "rendered": "S Monaco Pkwy & Evans Ave",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="transit icon"
                  className="transit-icon"
                  src="undefined"
/>,
                <div
                  className="line-id-container"
                  style={
                                    Object {
                                                      "background": "#0277bd",
                                                    }
                  }
>
                  <p
                                    className="line-id"
                  />
</div>,
              ],
              "className": "line-info-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "transit icon",
                  "className": "transit-icon",
                  "src": "undefined",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="line-id"
/>,
                  "className": "line-id-container",
                  "style": Object {
                    "background": "#0277bd",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "line-id",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Monaco Pkwy & 29th Ave",
              "className": "arrive-stop",
            },
            "ref": null,
            "rendered": "Monaco Pkwy & 29th Ave",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="step-time-data"
>
            <p
                        className="depart-time"
            >
                        5:51am
            </p>
            <p
                        className="arrive-time"
            >
                        6:07am
            </p>
</div>,
          <hr />,
          <div
            className="step-locations-container"
>
            <h3
                        className="depart-stop"
            >
                        S Monaco Pkwy & Evans Ave
            </h3>
            <div
                        className="line-info-container"
            >
                        <img
                                    alt="transit icon"
                                    className="transit-icon"
                                    src="undefined"
                        />
                        <div
                                    className="line-id-container"
                                    style={
                                                Object {
                                                            "background": "#0277bd",
                                                          }
                                    }
                        >
                                    <p
                                                className="line-id"
                                    />
                        </div>
            </div>
            <h3
                        className="arrive-stop"
            >
                        Monaco Pkwy & 29th Ave
            </h3>
</div>,
        ],
        "className": "transit-leg-container",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="depart-time"
>
                5:51am
</p>,
              <p
                className="arrive-time"
>
                6:07am
</p>,
            ],
            "className": "step-time-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5:51am",
                "className": "depart-time",
              },
              "ref": null,
              "rendered": "5:51am",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6:07am",
                "className": "arrive-time",
              },
              "ref": null,
              "rendered": "6:07am",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="depart-stop"
>
                S Monaco Pkwy & Evans Ave
</h3>,
              <div
                className="line-info-container"
>
                <img
                                alt="transit icon"
                                className="transit-icon"
                                src="undefined"
                />
                <div
                                className="line-id-container"
                                style={
                                                Object {
                                                                "background": "#0277bd",
                                                              }
                                }
                >
                                <p
                                                className="line-id"
                                />
                </div>
</div>,
              <h3
                className="arrive-stop"
>
                Monaco Pkwy & 29th Ave
</h3>,
            ],
            "className": "step-locations-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "S Monaco Pkwy & Evans Ave",
                "className": "depart-stop",
              },
              "ref": null,
              "rendered": "S Monaco Pkwy & Evans Ave",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="transit icon"
                    className="transit-icon"
                    src="undefined"
/>,
                  <div
                    className="line-id-container"
                    style={
                                        Object {
                                                            "background": "#0277bd",
                                                          }
                    }
>
                    <p
                                        className="line-id"
                    />
</div>,
                ],
                "className": "line-info-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "transit icon",
                    "className": "transit-icon",
                    "src": "undefined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="line-id"
/>,
                    "className": "line-id-container",
                    "style": Object {
                      "background": "#0277bd",
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "line-id",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "p",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Monaco Pkwy & 29th Ave",
                "className": "arrive-stop",
              },
              "ref": null,
              "rendered": "Monaco Pkwy & 29th Ave",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
